terraform/modules/emr/templates/emr/check_status.sh

```diff
@@ -1,70 +1,40 @@
-  then
-   break
-  else
-    echo "$DEPENDANT_DATA_PRODUCT is running... Still waiting."
-  fi
-  sleep ${INTERVAL}
-done
-
-if [[ ${DEPENDANT_DATA_PRODUCT_STATUS^^} != "COMPLETED" ]]
-then
- echo "PDM did not finish. Exiting as a Failure - Please check previous dependencies for errors."
- exit 1
-fi
-
-#Get the correlation_id of the latest PDM run
-  LATEST_COR_ID=$(hive -S -e "USE audit; SELECT correlation_id FROM data_pipeline_metadata_hive dp INNER JOIN (
-   SELECT MAX(to_date(dateproductrun)) AS run_date FROM data_pipeline_metadata_hive WHERE dataproduct = '${DEPENDANT_DATA_PRODUCT}') md
-   ON to_date(dp.dateproductrun) = md.run_date
-    WHERE dp.dataproduct = '${DEPENDANT_DATA_PRODUCT}' AND upper(dp.status) = 'COMPLETED';") 
+  #Read correlation id
+  LATEST_COR_ID=`cat ~/${DATA_SOURCE}_CORRELATION_ID.txt`
 
   #RUN_ID will not be set if the data product did not run after PDM has completed
-  RUN_ID=$(hive -S -e "USE audit; SELECT run_id from data_pipeline_metadata_hive WHERE dataproduct = '${DATA_PRODUCT}' AND upper(status) = 'COMPLETED' AND correlation_id = '${LATEST_COR_ID}';")
+  RUN_ID=$(hive -e "USE audit; SELECT run_id from data_pipeline_metadata_hive WHERE dataproduct = '${DATA_PRODUCT}' AND upper(status) = 'COMPLETED' AND correlation_id = '${LATEST_COR_ID}';")
+  echo "RUN_ID is ${RUN_ID} "
 
-  #Check for FAILED run
-  RUN_STATUS=$(hive -S -e "USE audit; SELECT status FROM data_pipeline_metadata_hive WHERE dataproduct = '${DATA_PRODUCT}' AND upper(status) = 'FAILED' AND correlation_id = '${LATEST_COR_ID}';")
+  #Get run status
+  RUN_STATUS=$(hive -e "USE audit; SELECT status FROM data_pipeline_metadata_hive WHERE dataproduct = '${DATA_PRODUCT}' AND correlation_id = '${LATEST_COR_ID}';")
+  echo "RUN_STATUS is ${RUN_STATUS} "
 
   if [  -z ${RUN_ID} ]
   then
-    echo "$DEPENDANT_DATA_PRODUCT has finished, Dependencies have finished - Proceeding with flow"
     echo "correlation_id is :"$LATEST_COR_ID
+    echo "Assigning the new Run Id to 1 "
     RUN_ID=1
-    echo "Run Id is: "$RUN_ID
-    ./update_dynamo.sh $LATEST_COR_ID "In-Progress" $DATA_PRODUCT $RUN_ID #Logging
+    echo "Calling update_dynamo.sh"
+    /opt/emr/repos/status_check/update_dynamo.sh $LATEST_COR_ID "In-Progress" $DATA_PRODUCT $RUN_ID #Logging
   elif [ ${RUN_STATUS^^} == "FAILED" ]
   then
-    echo "PDM is not running ............... "
-    echo "Re-Running $DATA_PRODUCT after FAILURE"
+    echo "Re-Running ${DATA_PRODUCT} after FAILURE"
     echo "correlation_id is :"$LATEST_COR_ID
+    echo "Incrementing the RUN_ID "
     RUN_ID=$(($RUN_ID+1)) #Increment run id for logging
-    echo "Run Id is: "$RUN_ID
-    ./update_dynamo.sh $LATEST_COR_ID "In-Progress" $DATA_PRODUCT $RUN_ID #Logging
+    echo "New Run Id is: "$RUN_ID
+    echo "Calling update_dynamo.sh"
+    /opt/emr/repos/status_check/update_dynamo.sh $LATEST_COR_ID "In-Progress" $DATA_PRODUCT $RUN_ID #Logging
   elif [ ${RUN_STATUS^^} == "IN-PROGRESS" ]
```

Comment:
Won't this check fail? We have been setting the value to 'In-Progress' in the dynamo parameters in this script and the comparison isn't case-insensitive.